lumicks/pylake/kymotracker/detail/peakfinding.py

```diff
@@ -62,28 +154,30 @@ def __init__(self, coordinates, time_points, peak_amplitudes):
             When no points are given or when `coordinates`, `time_points` and `peak_amplitudes`
             don't have the same number of elements.
         """
-        if len(time_points) == 0:
-            raise ValueError("You need to provide at least one time point")
-
         if any(len(time_points) != len(x) for x in (coordinates, peak_amplitudes)):
             raise ValueError(
                 f"Number of time points ({len(time_points)}), coordinates ({len(coordinates)}) and "
                 f"peak amplitudes ({len(peak_amplitudes)}) must be equal"
             )
 
         self.frames = []
-        max_frame = math.ceil(np.max(time_points))
-        for current_frame in np.arange(max_frame + 1):
-            (in_frame_idx,) = np.where(
-                np.logical_and(time_points >= current_frame, time_points < current_frame + 1)
-            )
-            self.frames.append(
-                self.Frame(
-                    coordinates[in_frame_idx],
-                    time_points[in_frame_idx],
-                    peak_amplitudes[in_frame_idx],
+        coordinates, time_points, peak_amplitudes = (
+            np.asarray(c) for c in (coordinates, time_points, peak_amplitudes)
+        )
+
+        if len(time_points) > 0:
```

Comment:
early return instead could save one level of nesting here

Reply:
nice catch